test/blackbox-tests/test-cases/build-bin.t

```diff
@@ -0,0 +1,22 @@
+Pform expansion should work in `exec` but also in `build`:
+
+  $ cat > dune-project << EOF
+  > (lang dune 3.10)
+  > (package (name e))
+  > EOF
+  $ cat > dune << EOF
+  > (executable (public_name e))
+  > EOF
+
+The executable just displays "Hello".
+
+  $ cat > e.ml << EOF
+  > let () = print_endline "Hello"
+  > EOF
+
+Making Dune build `e` should work:
+
+  $ dune build %{bin:e}
+  Error: File unavailable:
+  $TESTCASE_ROOT/../install/default/bin/e
```

Comment:
I think the issue here is that the path instead of `$TESTCASE_ROOT` (`Path.root`) should be the root of the context. E.g. `Context_name.build_dir $context`.

Reply:
Yes, this was pretty much it: #8594. The other issues need some looking into.